### ISSUE TITLE
Quite slow

### ISSUE THREAD

**Issue by Stay1444** [KuabeM/lcd-lcm1602-i2c#1]
I found this fork from the original library which had a bug that would only write the first character when calling `write_str`. This appears to fix it as I did (increasing the cursor and wriitng another character) the problem is that it's slow. Any ideas?

**Comment by KuabeM** [comment]
I actually noticed this as well but never came around to investigate this. I haven't used this crate in over a year I guess.

You could have a look at the code, there are a couple of sleeps that you might can get rid of/decrease the time: https://github.com/KuabeM/lcd-lcm1602-i2c/blob/master/src/lib.rs#L186.

I don't remember why they were necessary, but there was probably a reason.

**Comment by Stay1444** [comment]
Yeah, I found a fix to the two bugs. The first "bug" is that if you don't call clear on the LCD before writing a string it will only write the first character, always.

Then the other is not a bug but it's just that the delays present in the write4bits function can be reduced by a long shot.

This is my modified write function
```rs
    fn write4bits(&mut self, data: u8) -> Result<(), I2cError> {
        self.i2c.write(
            self.address,
            &[data | DisplayControl::DisplayOn as u8 | self.backlight_state as u8],
            10.ms().into(),
        )?;
        Ets::delay_us(500);
        self.i2c.write(
            self.address,
            &[DisplayControl::Off as u8 | self.backlight_state as u8],
            10.ms().into(),
        )?;
        Ets::delay_us(1500);
        Ok(())
    }
```

I removed the abstraction because I don't need them.

**Comment by KuabeM** [comment]
I had a look myself and was able to reduce the delay significantly, closed with ad13901

**Comment by Stay1444** [comment]
Nice! I'd also recommend to do an implicit clear inside the init function, since if you don't call clear after init it's broken and it won't draw more than one character per line.
